fe/free_energy.py

```diff
@@ -130,90 +74,42 @@ def __init__(self, mol, ff):
         self.ff = ff
         self.top = topology.BaseTopology(mol, ff)
 
-    # this can be used for both the solvent leg and the complex leg
-    def host_edge(self, lamb, host_system, host_coords, box, equil_steps=10000, prod_steps=100000):
+
+    def prepare_host_edge(self, ff_params, host_system, host_coords):
         """
-        Run equilibrium decoupling simulation at a given value of lambda in a host environment.
+        Prepares the host-edge system
 
         Parameters
         ----------
-        lamb: float [0, 1]
-            0 is the fully interacting system, and 1 is the non-interacting system
+        ff_params: tuple of np.array
+            Exactly equal to bond_params, angle_params, proper_params, improper_params, charge_params, lj_params
 
         host_system: openmm.System
-            OpenMM System object to be deserialized. The host can be simply a box of water, or a fully
-            solvated protein
-
-        host_coords: np.array of shape [..., 3]
-            Host coordinates, in nanometers. It should be properly minimized and not have clashes
-            with the ligand coordinates.
-
-        box: np.array [3,3]
-            Periodic boundary conditions, in nanometers.
-
-        equil_steps: float
-            Number of steps to run equilibration. Statistics are not gathered.
-
-        prod_steps: float
-            Number of steps to run production. Statistics are gathered.
+            openmm System object to be deserialized
 
+        host_coords: np.array
+            Nx3 array of atomic coordinates
 
         Returns
         -------
-        float, float
-            Returns a pair of average du_dl values for bonded and nonbonded terms.
+        4 tuple
+            bound_potentials, vjp_fns_and_handles, combined_masses, combined_coords
```

Comment:
update to match return types
```suggestion
            final_potentials, final_params, combined_masses, combined_coords
```